File: libusbcamera/src/main/java/com/rain/uvc/state/CameraPreviewFormat.java
package com.rain.uvc.state;

/**
 * 当前支持的格式
 */
public enum CameraPreviewFormat {
    BGR(0), YUY2(1), NV21(2), NV12(3), RGB(5), MJPEG(6), JPEG(7);
    private final int value;

    CameraPreviewFormat(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static CameraPreviewFormat valueToFormatMode(int format) {
        switch (format) {
            case 0:
                return CameraPreviewFormat.BGR;
            case 2:
                return CameraPreviewFormat.NV21;
            case 3:
                return CameraPreviewFormat.NV12;
            case 5:
                return CameraPreviewFormat.RGB;
            case 6:
                return CameraPreviewFormat.MJPEG;
            case 7:
                return CameraPreviewFormat.JPEG;
            default:
                return CameraPreviewFormat.YUY2;
        }
    }
}

File: libusbcamera/src/main/java/com/rain/uvc/parameters/CameraPreviewFormat.kt
package com.rain.uvc.parameters

/**
 * @author yuan
 * @createTime: 2026/2/2
 * @des 当前支持的预览格式
 */
sealed class CameraPreviewFormat(val value: Int) {
	companion object {
		@JvmStatic
		fun valueToFormatMode(format: Int): CameraPreviewFormat {
			return when (format) {
				0 -> BGR;
				2 -> return NV21
				3 -> return NV12
				5 -> return  RGB
				6 -> return  MJPEG
				7 -> return  JPEG
				else -> return  YUY2
			}
		}
	}
	
	data object BGR : CameraPreviewFormat(0)
	data object YUY2 : CameraPreviewFormat(1)
	data object NV21 : CameraPreviewFormat(2)
	data object NV12 : CameraPreviewFormat(3)
	data object RGB : CameraPreviewFormat(5)
	data object MJPEG : CameraPreviewFormat(6)
	data object JPEG : CameraPreviewFormat(7)
}